doc/manuals/admin/extra/ha.md

```diff
@@ -0,0 +1,441 @@
+
+    ```bash
+    $ nano /etc/yum.repos.d/fiware.repo
+
+    [fiware]
+    name=Fiware Repository
+    baseurl=http://repositories.lab.fiware.org/repo/rpm/$releasever
+    gpgcheck=0
+    enabled=1
+
+    $ yum install contextBroker
+    ```
+
+1. Configure the Context Broker by listing all the mongodb hosts and the name
+   of the replica set
+
+    ```bash
+    $ nano /etc/sysconfig/contextBroker
+
+    BROKER_DATABASE_HOST=mongo1,mongo2,mongo3
+    BROKER_DATABASE_RPLSET=contextBroker
+    ```
+
+1. Start the Context Broker
+
+    ```bash
+    $ /etc/init.d/contextBroker start
+    ```
+
+1. Test the Context Broker
+
+    ```bash
+    $ curl localhost:1026/v2/entities -s -S --header 'Accept: application/json' | python -mjson.tool
+    ```
+
+### HA Proxy Configuration
+
+1. Configure the Firewall to block access from outside the local network except
+   for SSH and for the 1026 that will be used to load balance Context Broker
+   traffic
+
+    ```bash
+    $ nano /etc/iptables.rules
+
+    *filter
+    :INPUT DROP [0:0]
+    :FORWARD ACCEPT [0:0]
+    :OUTPUT ACCEPT [0:0]
+    :LOGGING - [0:0]
+    #COMMIT
+    -A INPUT -m state --state RELATED,ESTABLISHED -j ACCEPT
+    -A INPUT -p icmp -s 10.0.64.0/25 -j ACCEPT
+    -A INPUT -i lo -j ACCEPT
+    -A INPUT -m state --state NEW -m tcp -p tcp --dport 1026 -j ACCEPT
+    -A INPUT -p tcp -m state --state NEW -m tcp --dport 22 -j ACCEPT
+    COMMIT
+
+    $ iptables-restore < /etc/iptables.rules
+    ```
+
+1. Configure the hosts file to include local network hostname resolution
+
+    ```bash
+    $ nano /etc/hosts
+
+    10.0.64.32   mongo1
+    10.0.64.33   mongo2
+    10.0.64.34   mongo3
+
+    10.0.64.32   mongo1.lab.fiware.org
+    10.0.64.33   mongo2.lab.fiware.org
+    10.0.64.34   mongo3.lab.fiware.org
+    10.0.64.35   contextbroker1.lab.fiware.org
+    10.0.64.36   contextbroker2.lab.fiware.org
+    10.0.64.37   contextbroker3.lab.fiware.org
+    10.0.64.38   proxy1.lab.fiware.org
+    10.0.64.39   proxy2.lab.fiware.org
+    10.0.64.40   proxy3.lab.fiware.org
+    ```
+
+1. Install the HA Proxy
+
+    ```bash
+    $ sudo apt-get install haproxy
+    ```
+
+1. Configure the HA Proxy
+
+    ```bash
+    $ nano /etc/haproxy/haproxy.cfg
+
+    frontend www
+            bind *:1026
+            default_backend ctx_pool
+
+    backend ctx_pool
+            balance roundrobin
+            mode http
+            server ctx1 contextbroker1.lab.fiware.org:1026 check
+            server ctx2 contextbroker2.lab.fiware.org:1026 check
+            server ctx3 contextbroker3.lab.fiware.org:1026 check
+    ```
+
+1. Enable the automatic restart of HA Proxy on reboot
+
+    ```bash
+    $ nano /etc/default/haproxy
+
+    ENABLED=1
+    ```
+
+1. Start the HA Proxy
+
+    ```bash
+    $ service haproxy reload
+    ```
+
+1. Test the HA Proxy
+
+    ```bash
+    $ curl localhost:1026/v2 -s -S --header 'Accept: application/json' | python -mjson.tool
```

Comment:
Typically the operation to test CB is /version, which is much simpler (no need of adding Accept header nor piping python for rendering):

```
curl localhost:1026/v2
```